basic-usage/create-mocks-by-example.md

```diff
@@ -1,346 +1,366 @@
-	
-	  #### __[C#]__
-	
-	  {{region CreateMocksByExample#ArrangingOnlyReturnsNewSyntaxis}}
-	    [TestMethod]
-        public void SampleTest()
-        {
-            var conn = Mock.CreateLike<IConnection>(me => me.Driver == "MSSQL" && me.Open(Arg.AnyString) == true);
-            Assert.IsTrue(conn.Open(@".\SQLEXPRESS"));
-        }
-	  {{endregion}}
-	
-	  #### __[VB]__
-	
-	  {{region CreateMocksByExample#ArrangingOnlyReturnsNewSyntaxis}}
-	    <TestMethod> _
-        Public Sub SampleTest()
-            Dim conn = Mock.CreateLike(Of IConnection)(Function([me]) [me].Driver = "MSSQL" AndAlso [me].Open(Arg.AnyString) = True)
-            Assert.IsTrue(conn.Open(".\SQLEXPRESS"))
-        End Sub
-	  {{endregion}}
-	
-	 However, if we want to add expectations to `Open()`, we cannot with the above syntax. We have to use the old syntax. We also can’t define the return value with the new syntax and then the expectations with `Mock.Arrange()`. We have to use the old one for the entire arrangement: 
-	
-	  #### __[C#]__
-	
-	  {{region CreateMocksByExample#ArrangingOnlyReturnsOldSyntax}}
-	    [TestMethod]
-        public void ShouldUseCreateLikeAlongWithStandartArrangements()
-        {
-            var conn = Mock.CreateLike<IConnection>(me => me.Driver == "MSSQL");
-            Mock.Arrange(() => conn.Open(Arg.AnyString)).Returns(true).MustBeCalled();
-            Assert.IsTrue(conn.Open(@".\SQLEXPRESS"));
-            Mock.Assert(conn);
-        }
-	  {{endregion}}
-	
-	  #### __[VB]__
-	
-	  {{region CreateMocksByExample#ArrangingOnlyReturnsOldSyntax}}
-	    <TestMethod> _
-        Public Sub SampleTestusingMockCreate()
-            Dim conn = Mock.CreateLike(Of IConnection)(Function([me]) [me].Driver = "MSSQL")
-            Mock.Arrange(Function() conn.Open(Arg.AnyString)).Returns(True).MustBeCalled()
-            Assert.IsTrue(conn.Open(".\SQLEXPRESS"))
-            Mock.Assert(conn)
-        End Sub
-	  {{endregion}}
-
-
-* __At any time, for any mock, you can add additional arrangements using the new syntax by calling__
+---
+title: Create and Arrange at the Same Time
+page_title: Create and Arrange Mocks at the Same Time | JustMock Documentation
+description: Create and Arrange Mocks at the Same Time
+previous_url: /basic-usage-create-mocks-by-example.html
+slug: justmock/basic-usage/create-mocks-by-example
+tags: create,mock,arrange,createlike
+published: True
+position: 14
+---
+
+# Create and Arrange Mocks at the Same Time
+
+The built-in feature for creating mocks by specific arrangement saves time when it comes to tiresome set up of arrangements. This functionality allows you to create mocks of a certain class (the system under test) and to arrange their behavior at the same time through the **Mock.CreateLike** method. 
+
+>noteThis feature is available in both - **JustMock** and **JustMock Lite**. In **JustMock**, with the profiler enabled, you can create arrangements that require elevated mocking.
```

Comment:
This note bothers me. I prefer only the articles that require the profiler to have an explicit message about it. All others do not need that as it will work in any case, and such a message can only scare the reader.